### PR TITLE
Implement support for dpnp.arange()

### PR DESCRIPTION
**Pull request by chudur-budur**
WIP for implementing overloads of `dpnp.arange()`.
This PR also implements dispatch vector to store template instantiated sycl kernels for basic tensor ops.

- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to an issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

**Comment by chudur-budur** [comment]
Deleted by accident.

File: .clang-format [hunk 1]
```diff
@@ -5,6 +5,7 @@ AlignEscapedNewlines: Right
 AllowAllParametersOfDeclarationOnNextLine: false
 BinPackParameters: false
 BreakBeforeBraces: Custom
+SortIncludes: false
```
Comment (diptorupd) [inline .clang-format:8]: Why do we need this change? In general I do not like these type of changes as it indicates that headers should be in a specific order and any inadvertent change will introduce issues.

File: numba_dpex/core/runtime/kernels/tensor/include/typeutils.hpp [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+#ifndef __TYPES_HPP__
+#define __TYPES_HPP__
+
+#include <cstdlib>
+#include <complex>
+#include <exception>
+#include <utility>
+#include <string>
+#include <cstdint>
+// #include <cxxabi.h> // this is gcc specific, not supported on windows
```
Comment (diptorupd) [inline numba_dpex/core/runtime/kernels/tensor/include/typeutils.hpp:14]: leftover cleanup?

Comment (chudur-budur) [reply numba_dpex/core/runtime/kernels/tensor/include/typeutils.hpp:14]: This is needed for the `demangle()` function, which is very useful for debugging this kind of templated code.

File: numba_dpex/core/runtime/context.py [hunk 3]
```diff
@@ -21,7 +21,9 @@ def _check_null_result(func):
         @functools.wraps(func)
         def wrap(self, builder, *args, **kwargs):
             memptr = func(self, builder, *args, **kwargs)
-            msg = "USM allocation failed. Check the usm_type and queue."
+            msg = "In {0:s}.wrap(). USM allocation failed at {1:s}(). Check the usm_type and queue.".format(
```
Comment (diptorupd) [inline numba_dpex/core/runtime/context.py:24]: Please use f-string and look out for 80 char libe width .

File: numba_dpex/core/runtime/kernels/tensor/include/typeutils.hpp (hunk 2 above)
Comment (diptorupd) [reply numba_dpex/core/runtime/kernels/tensor/include/typeutils.hpp:14]: We should not have commented out code in our sources. We recently added a flake8 check to disallow it explicitly for the Python code.

File: .clang-format (hunk 1 above)
Comment (chudur-budur) [reply .clang-format]: Removed.

File: numba_dpex/core/runtime/kernels/tensor/include/typeutils.hpp (hunk 2 above)
Comment (chudur-budur) [reply numba_dpex/core/runtime/kernels/tensor/include/typeutils.hpp]: Fixed.

File: numba_dpex/core/runtime/context.py (hunk 3 above)
Comment (chudur-budur) [reply numba_dpex/core/runtime/context.py]: Fixed.

**Comment by diptorupd** [comment]
Thank you for the exploratory work @chudur-budur!

As discussed, we will in future explore adding support for these constructors using the `overload` functionality and using the kernel api. As such the current PR is out of scope and I am closing it.

**Pull request closed without merging**